aea/test_tools/test_cases.py

```diff
@@ -0,0 +1,289 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+import os
+import shutil
+import subprocess  # nosec
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from aea.cli import cli
+from aea.configurations.base import PublicId
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.exceptions import AEATestingException
+
+from tests.common.click_testing import CliRunner
+from tests.conftest import AUTHOR, CLI_LOG_OPTION
+
+
+class AEATestCase:
+    """Test case for AEA end-to-end tests."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        cls.runner = CliRunner()
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+
+        # add packages folder
+        packages_src = os.path.join(cls.cwd, "packages")
+        packages_dst = os.path.join(cls.t, "packages")
+        shutil.copytree(packages_src, packages_dst)
+
+        os.chdir(cls.t)
+
+    @classmethod
+    def teardown_class(cls):
+        """Teardowm the test."""
+        os.chdir(cls.cwd)
+        try:
+            shutil.rmtree(cls.t)
+        except (OSError, IOError):
+            pass
+
+    def disable_ledger_tx(self, vendor_name, item_type, item_name):
+        """
+        Disable ledger tx by modifying item yaml settings.
+        Run from agent's directory and only for item with present strategy is_ledger_tx setting.
+
+        :param vendor_name: str vendor name.
+        :param item_type: str item type.
+        :param item_name: str item name.
+
+        :return: None
+        """
+        json_path = "vendor.{}.{}s.{}.models.strategy.args.is_ledger_tx".format(
+            vendor_name, item_type, item_name
+        )
+        self.run_cli_command("config", "set", json_path, False)
+
+    def disable_aea_logging(self):
+        """
+        Disable AEA logging of specific agent.
+        Run from agent's directory.
+
+        :return: None
+        """
+        config_update_dict = {
+            "agent.logging_config.disable_existing_loggers": "False",
+            "agent.logging_config.version": "1",
+        }
+        for path, value in config_update_dict.items():
+            self.run_cli_command("config", "set", path, value)
+
+    def run_cli_command(self, *args):
```

Comment:
This should be a private method?

Reply:
There could be possible use cases that may need this method outside.

Reply:
Ok